filament/backend/src/vulkan/VulkanPipelineCache.cpp

```diff
@@ -46,21 +46,13 @@ static VkShaderStageFlags getShaderStageFlags(utils::bitset64 key, uint16_t bind
     return flags;
 }
 
-utils::bitset64
-static getPipelineLayoutKey(const Program::SamplerGroupInfo& samplerGroupInfo) noexcept {
-    utils::bitset64 key = {};
-    for (uint32_t binding = 0; binding < Program::SAMPLER_BINDING_COUNT; ++binding) {
-        const auto& stageFlags = samplerGroupInfo[binding].stageFlags;
-        for (const auto& sampler : samplerGroupInfo[binding].samplers) {
-            if (any(stageFlags & ShaderStageFlags::VERTEX)) {
-                key.set(sampler.binding * 2 + 0);
-            }
-            if (any(stageFlags & ShaderStageFlags::FRAGMENT)) {
-                key.set(sampler.binding * 2 + 1);
-            }
-        }
+void VulkanPipelineCache::setUsageFlags(UsageFlags* key, uint16_t binding, ShaderStageFlags flags) {
```

Comment:
I think it would be better to call it `getUsageFlags` and have it return a `UsageFlags` that you *or* with the one you already have.

Reply:
Pushed a new commit for this.